### PR TITLE
feat(email): add email offer upgrade end points

### PR DESCRIPTION
**Pull request by radireddy**
Added email order end points

### CODE REVIEW

File: src/api/order/email/domain/order-email-domain.v6.service.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+angular.module('ovh-api-services').service('OvhApiOrderDomainV6', ($resource, $cacheFactory) => {
```
Comment (marie-j) [inline src/api/order/email/domain/order-email-domain.v6.service.js]:
```suggestion
angular.module('ovh-api-services').service('OvhApiOrderEmailDomainV6', ($resource, $cacheFactory) => {
```
else there could be some conflits with the `/order/domain` route

File: src/api/order/email/order-email.service.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+angular.module('ovh-api-services').service('OvhApiOrderEmail', ($injector) => ({
+  Domain() {
+    return $injector.get('OvhApiOrderDomain');
```
Comment (FredericEspiau) [inline src/api/order/email/order-email.service.js]: You meant `OvhApiOrderEmailDomain` I think 🙂

File: src/api/order/email/domain/order-email-domain.v6.service.js (hunk 1 above)
Comment (FredericEspiau) [inline src/api/order/email/domain/order-email-domain.v6.service.js]:
```suggestion
angular.module('ovh-api-services').service('OvhApiOrderEmailDomainV6', ($resource, $cacheFactory) => {
```

File: src/api/order/email/domain/order-email-domain.service.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+angular.module('ovh-api-services').service('OvhApiOrderDomain', ($injector) => ({
```
Comment (FredericEspiau) [inline src/api/order/email/domain/order-email-domain.service.js]:
```suggestion
angular.module('ovh-api-services').service('OvhApiOrderEmailDomain', ($injector) => ({
```

File: src/api/order/email/domain/order-email-domain.service.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+  v6() {
+    return $injector.get('OvhApiOrderDomainV6');
```
Comment (FredericEspiau) [inline src/api/order/email/domain/order-email-domain.service.js]:
```suggestion
    return $injector.get('OvhApiOrderEmailDomainV6');
```

**Comment by FredericEspiau** [comment]
Why are you creating methods for deprecated routes ?

https://api.ovh.com/console/#/order/email/domain/%7Bdomain%7D#GET

File: src/api/order/email/domain/order-email-domain.v6.service.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      orderEmailDomain.resetQueryCache();
+      return response.data;
+    },
+  };
+
+  const orderEmailDomain = $resource('/order/email/domain/:domain', {
+    domain: '@domain',
+  }, {
+    query: { method: 'GET', cache: queryCache, isArray: true },
+    getDuration: {
```
Comment (marie-j) [inline src/api/order/email/domain/order-email-domain.v6.service.js]:
```suggestion
    getOption: {
```
would be better as tehre is no `/duration` in the route and you have to pass an `option` parameter

**Pull request merged**